app/Widgets/Hero_Slider_Widget.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+namespace Boilerplate\Theme\Widgets;
+
+use JustCoded\WP\Framework\Page_Builder\v25\Page_Builder_Widget;
+use JustCoded\WP\Framework\Web\View;
+
+/**
+ * Class Hero_Slider_Widget
+ * Custom image wordpress widget based on siteorigin widgets bundle
+ */
+class Hero_Slider_Widget extends Page_Builder_Widget {
+	/**
+	 * Hero_Slider_Widget constructor.
+	 */
+	function __construct() {
+		parent::__construct(
+			'hero-image',
+			__( 'Slider image widget' ),
+			array(
+				'description' => __( 'Slider image widget.' ),
+				'has_preview' => true,
+			),
+			array(),
+			false,
+			''
+		);
+	}
+
+	/**
+	 * Return a list of images to be shown as preview.
+	 * By default it's widget {id_base}.png with $this->name
+	 *
+	 * @return array
+	 */
+	public function get_preview_images() {
+		return array(
+			'slider-image.png' => 'Preview of the slider widget',
+		);
+	}
+
+	/**
+	 * Form fields configuration
+	 *
+	 * @return array
+	 */
+	function get_widget_form() {
+		return
+			array(
+				'images'      => array(
+					'type'       => 'repeater',
+					'label'      => __( 'Images', 'so-widgets-bundle' ),
+					'item_name'  => __( 'Image', 'so-widgets-bundle' ),
+					'item_label' => array(
+						'selector'     => "[name*='title']",
+						'update_event' => 'change',
+						'value_method' => 'val',
+					),
+
+					'fields' => array(
+						'image'       => array(
+							'type'  => 'media',
+							'label' => __( 'Image', 'so-widgets-bundle' ),
+						),
+						'title'       => array(
+							'type'  => 'text',
+							'label' => __( 'Image title', 'so-widgets-bundle' ),
+						),
+						'description' => array(
+							'type'  => 'text',
+							'label' => __( 'Description', 'so-widgets-bundle' ),
+						),
+
+						'url'        => array(
+							'type'  => 'link',
+							'label' => __( 'URL', 'so-widgets-bundle' ),
+						),
+						'new_window' => array(
+							'type'    => 'checkbox',
+							'default' => false,
+							'label'   => __( 'Open in new window', 'so-widgets-bundle' ),
+						),
```

Comment:
It's better to add fields like:
Button Text
Button Link

instead of url and new window.

Reply:
added